agent/consul/federation_state_replication.go

```diff
@@ -4,11 +4,28 @@ import (
 	"context"
 	"fmt"
 	"sort"
+	"strings"
 	"time"
 
 	"github.com/hashicorp/consul/agent/structs"
 )
 
+const federationStatsNotSupportedMessage = "Not all servers in the datacenter support federation states - preventing replication"
+
+var (
+	federationStatesNotSupported = federationStatesNotSupportedError{}
+)
+
+type federationStatesNotSupportedError struct{}
+
+func (e federationStatesNotSupportedError) Error() string {
+	return federationStatsNotSupportedMessage
```

Comment:
Is this type being created so that it can be compared? For constant errors I believe the `==` operator works (https://play.golang.org/p/N945vkiTk5S) and is the usual way of comparing them.  https://golang.org/pkg/errors/#Is also works with constants and uses the `==` to compare them.

Reply:
I found the lone caller between the error being created and the error being inspected and swapped the `fmt.Errorf` directive from `%v` to `%w` so that `errors.Is()` actually works now.